src/components/Label/Label.tsx

```diff
@@ -43,11 +68,46 @@ class Label extends UIComponent<any, any> {
 
   static variables = labelVariables
 
+  handleIconOverrides = predefinedProps => {
+    const { onIconClick, iconPosition, content } = this.props
+    const { onClick, variables, xSpacing } = predefinedProps
+
+    return {
+      onClick: e => {
+        _.invoke(predefinedProps, 'onClick', e)
+        _.invoke(this.props, 'onIconClick', e, this.props)
+      },
+      ...((onClick || onIconClick) && { tabIndex: '0' }),
+      ...((!variables || !variables.color) && { variables: { color: Label.variables().color } }),
```

Comment:
The overrideProps functino is invoked inside the create factory, so I don't see a way to provide the additional labelVariables from here. The only argument it gets are the predefined props. I can however fix  the assumption that the variables are object inside the props of the icon shorthand with the following changes inside the handleOverrideProps like this:

```
  handleIconOverrides = (predefinedProps) => {
    const { onIconClick, iconPosition, content } = this.props
    const { onClick, variables, xSpacing } = predefinedProps

    // evaluate the variables inside the props (supporting variables: () => ({color: 'red'}) and variables: {color: 'red'})
    const evaluatedVariables = callable(variables)()

    return {
      onClick: e => {
        _.invoke(predefinedProps, 'onClick', e)
        _.invoke(this.props, 'onIconClick', e, this.props)
      },
      ...((onClick || onIconClick) && { tabIndex: '0' }),
      ...((!evaluatedVariables || !evaluatedVariables.color) && { variables: { color: Label.variables().color } }),
      ...(!xSpacing && {
        xSpacing: !content ? 'none' : iconPosition === 'end' ? 'before' : 'after',
      }),
    }
  }
```

What do you think @kuzhelov ?

Reply:
yes, that would be great - this will save us in situations when variables are provided as plain object. Would rather rename `evaluatedVariables` to `labelVariables` - it would be much easier to understand the next lines of code then. Other than that - absolutely agree, great suggestion!

Reply:
Got, it will do that. And yes, we still need a way to provide the variables together with the site variables inside the overrideProps function, we should have this is mind.

Reply:
Actually, the latest version of the method would look like this:

```
  handleIconOverrides = (predefinedProps) => {
    const { onIconClick, iconPosition, content, variables: labelPropVariables } = this.props
    const { onClick, variables, xSpacing } = predefinedProps

    const iconVariables = callable(variables)()
    const labelVariables = labelPropVariables ? callable(labelPropVariables)() : callable(Label.variables)()

    return {
      onClick: e => {
        _.invoke(predefinedProps, 'onClick', e)
        _.invoke(this.props, 'onIconClick', e, this.props)
      },
      ...((onClick || onIconClick) && { tabIndex: '0' }),
      ...((!iconVariables || !iconVariables.color) && { variables: { color: labelVariables.color } }),
      ...(!xSpacing && {
        xSpacing: !content ? 'none' : iconPosition === 'end' ? 'before' : 'after',
      }),
    }
  }
```
This way, I am handling the variables inside the label component, the default variables in the Label, as well as the variables inside the icon shorthand. This seems to work with every combination of variables provided in both components.